mesonbuild/scripts/tags.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2019 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import os
+
+
+def popen_list_files(srcdir_name):
+    if os.path.exists('.git'):
+        ls = subprocess.Popen(['git', 'ls-tree', '-r', '--name-only', 'HEAD'],
+                              stdout=subprocess.PIPE)
+    else:
+        ls = subprocess.Popen(['find', srcdir_name, '-name', '.*', '-prune',
+                               '-o', '-type', 'f', '-print'],
+                              stdout=subprocess.PIPE)
+    return ls
+
+
+def cscope(srcdir_name):
+    ls = popen_list_files(srcdir_name)
+    ls = subprocess.Popen(['sed', 's/^/"/;s/$/"/'], stdin=ls.stdout,
```

Comment:
Use of shell commands is not permitted as they are not portable. Use Python's functionality instead of `sed` or `find`.

Reply:
The `find` shell command can be done using [Pathlib.Path.iterdir()](https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir) for example

Reply:
done. The new version will be slightly less efficient, as find is able to skip subtrees, and it no longer uses subprocesses and pipes..